File: OOP in Java/Calculator/src/main/java/org/Calculadora.java
package org;

public class Calculadora {
    public double soma(double valor1, double valor2){
        return valor1 + valor2;
    }

    public double subtracao(double valor1, double valor2){
        return valor1 - valor2;
    }

    public double multiplicacao(double valor1, double valor2){
        return valor1 * valor2;
    }

    public double divisao(double valor1, double valor2){
        return valor1 / valor2;
    }

    public double raiz(double valor1){
        return Math.sqrt(valor1);
    }

    public double exponenciacao(double valor1, double expoente){
        return Math.pow(valor1, expoente);
    }

    public double bhaskara(double a, double b){
        return (-b) / a;
    }
}

File: Modular Programming/Calculator/src/calculadora.js
export const OPERACAO_INVALIDA = 'OPERACAO_INVALIDA'

export const calcular = (operacao, valores) => {
  let valor = valores[0];
  if (operacao == "soma") {
    for (let i = 1; i < valores.length; i++){
      valor += valores[i];
    }
  }

  else if (operacao == "subtracao") {
    for (let i = 1; i < valores.length; i++) {
      valor -= valores[i];
    }
  }

  else if (operacao == "multiplicacao") {
    for (let i = 1; i < valores.length; i++) {
      valor *= valores[i];
    }
  }

  else if (operacao == "divisao") {
    for (let i = 1; i < valores.length; i++) {
      valor /= valores[i];
    }
  }
  else {
    return OPERACAO_INVALIDA;
  }

  return valor;
}
